LukeExercise2.py

```diff
@@ -1,28 +1,44 @@
-from os import path
-import pandas as pd
-
-curr_dir = path.abspath(path.dirname(__file__))
-# Use the following two lines to interactively test the script
-# in iPython
-#os.chdir("C:/Users/luked/Documents/Code/pyCEP/DeveloperAssessment")
-#curr_dir = os.getcwd()
-
-ex2_input_dir = path.join(curr_dir, "exercise2", "input")
-
-# Read in the data file
-mean = pd.read_csv(path.join(ex2_input_dir, "mean.csv"))
-
-# This works in the current version of pandas but not the old one
-# we use for CREPE (because rank() didn't have the "pct" arg).
-# pct = mean.drop("fdntext", axis=1).rank(pct=True)
-
-# This way of calculating works in the old version of pandas
-rank = mean.drop("fdntext", axis=1).rank(ascending=False)
-fdn_count = len(mean["fdntext"])
-pct = 1 - rank / fdn_count
-
-pct.insert(0, "fdntext", mean["fdntext"])
-pct = pct.set_index("fdntext")
-
-# Output a csv with these percentiles
-pct.to_csv("pct.csv")
\ No newline at end of file
+from os import path
+import pandas as pd
+from scipy.stats import percentileofscore
+
+# Use the following lines to test the script in iPython
+#import os
+#os.chdir("Documents/Code/DeveloperAssessment")
+#curr_dir = os.getcwd()
+
+
+def pct_rank(data, score):
+    if pd.isnull(score):
+        return None
+    else:
+        return percentileofscore(data, score, kind="mean")
+
+
+def main():
+    curr_dir = path.abspath(path.dirname(__file__))
+    ex2_input_dir = path.join(curr_dir, "exercise2", "input")
+
+    # Read in the data file
+    mean = pd.read_csv(path.join(ex2_input_dir, "mean.csv"))
+    pct = mean

```

Comment:
This line is fine in this case, but to save memory operations, with this line, Python treats `pct` as an alias for `mean`, so say if you set `pct = None`, `mean` would be None too.

If you need `mean` for future use, you can use: `pct = mean.copy()` so that you can operate on `pct` while still have `mean` as original data for comparison, other operations, etc.

Not something to worry about for now. You'll develop an intuition for when to copy vs. when to write directly